services/st_util/StBootloaderCommunicatorUart.cpp

```diff
@@ -136,7 +138,8 @@ namespace services
 
     void StBootloaderCommunicatorUart::WriteMemory(uint32_t address, infra::ConstByteRange data, const infra::Function<void()>& onDone)
     {
-        really_assert(data.size() <= 255);
+        really_assert(data.size() <= 256);
+        really_assert(data.size() % 4 == 0);
```

Comment:
These kind of conditions need to be documented at the interface, otherwise they're Liskov substitution failures